src/main/java/com/redhat/devtools/intellij/lsp4ij/internal/CancellationSupport.java

```diff
@@ -10,10 +10,10 @@
  ******************************************************************************/
 package com.redhat.devtools.intellij.lsp4ij.internal;
 
-import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.CancellationException;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CopyOnWriteArrayList;

```

Comment:
I think we can use a simple arraylist

Reply:
copyonwritearraylist should not impact performance, as the list will stay small. At least it'll provide the guarantee there will be no concurrent modifications exception, how unlikely that be

Reply:
OK lets keep like this